### DISCUSSION TITLE
coupling problem

### DISCUSSION THREAD

**Discussion by zahidhasan83** [idaholab/moose#28104] [Q&A Modules: Phase field]
Dear all,
I am trying to simulate two different physics with two different input files using phase field module. However, the two different material physics are dependent on each other. Therefore, I am planning to do a coupled simulation. The computation will look like:
1. First run input1.i file to solve the first physics and get the output file (input1.e) with grain structure and grain boundaries.
2. Next, import the grain structure and boundaries from input1.e file to input2.i file and solve the second physics. Get the input2.e file from this step with output grain structure and grain boundaries.
3. Following that feed the grain structure and grain boundaries from input2.e file in the input1.i file and solve the first physics again. 

These simulation steps will repeat for a certain time. 
Would you be kind to let me know, how to do such computation in MOOSE? Do I need to use a Multiapp system for that? If so which Multiapp system will be the most appropriate for these problem? 

I would appreciate your advice.
Thanks, and best regards,
Zahid

**Comment by Rahim-Habibi** [comment]
Hi, 
I would suggest "TransientMultiApp', you can control the apps via "execute_on" and "sub_cycling".
OR 
Another option would be to apply "restart" to feed the parent/child at each step by child/parent output. 
Rahim

**Reply by zahidhasan83** [reply]
@Rahim-Habibi Dear Rahim, since I am exchanging the grain structure and grain boundaries between two simulations, do you think the following input in the parent file makes sense for a TransientMultiApp:
"[MultiApps]
  [sub]
    type = TransientMultiApp
    app_type = MooseTestApp
    positions = '0 0 0'
    input_files = 'recr-test-2.i'
    clone_parent_mesh = true
    execute_on = 'timestep_end'
  []
[]
[Transfers]
  [bnds_from_sub]
    type = MultiAppShapeEvaluationTransfer
    from_multi_app = sub
    source_variable = bnds
    variable = bnds
    execute_on = 'timestep_end'
  []
  [gr0_from_sub]
    type = MultiAppShapeEvaluationTransfer
    from_multi_app = sub
    source_variable = gr0
    variable = gr0
    execute_on = 'timestep_end'
  []
[bnds_to_sub]
    type = MultiAppShapeEvaluationTransfer
    to_multi_app = sub
    source_variable = bnds
    variable = bnds
    execute_on = 'timestep_end'
  []
  [gr0_to_sub]
    type = MultiAppShapeEvaluationTransfer
    to_multi_app = sub
    source_variable = gr0
    variable = gr0
    execute_on = 'timestep_end'
  []
[]"

Other queries:
I am using the identical computational domain and scale for both simulations. Therefore, positions = '0 0 0' is correct? If clone_parent_mesh = true, will the second.i use the output mesh from the first.i?

Thanks, and best regards,
Zahid

**Reply by Rahim-Habibi** [reply]
Hi, 
Yes, it seems OK. 
Regarding your second query, "clone_parent_mesh" is about the mesh and initilizes the apps by same mesh when it is true.  Generally you can define the mesh either in child app or parant app using this option.
When you want to run child app in spacfic position(s) you need to set position otherwise just leave the setting as it is (I think so). 
To have control on the outputs, check "Transfers" objects and use the proper one based on your problem. 
Rahim

**Comment by GiudGiud** [comment]
Hello

Is this a transient and you are solving physics1 then physics2 every time step?
In this case use a TransientMultiapp

Or are you looking for a steady state equilibrium solution?
In this case use a FullSolveMultiapp

You can use Transfers to pass the solution between the two solves
There is a tutorial on this here:
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial02_multiapps/presentation/index.html#/

Guillaume

**Reply by zahidhasan83** [reply]
@GiudGiud Appreciate.
